onnxruntime/python/tools/transformers/large_model_exporter.py

```diff
@@ -0,0 +1,282 @@
+        onnx_inputs[idx] = user_inputs[0][idx]
+    for key, value in user_inputs[1].items():
+        idx = input_keys.index(key)
+        onnx_inputs[idx] = value
+    for idx, (key, value) in enumerate(zip(input_keys, onnx_inputs)):
+        if type(value) is torch.Tensor:
+            value.to(model.device)
+        # Didn't touch past_key_value now, please change it if you want
+        if "use_cache" in key:
+            onnx_inputs[idx] = False
+
+    return input_keys, tuple(onnx_inputs)
+
+
+@torch.no_grad()
+def export_onnx(hf_model: str, onnx_path_str: str):
+    """
+    do export
+    model: torch model
+    onnx_path: where the onnx model saved to
+    sample_inputs_tp: inputs for torch model
+    """
+    onnx_model_name, model, sample_inputs_tp = model_prepare(hf_model)
+    total_mem_per_cpu = torch.cuda.get_device_properties(0).total_memory / 1024 / 1024
+
+    print("Model_Size", get_model_size(model))
+    print("total_mem_per_cpu=", total_mem_per_cpu)
+    if get_model_size(model) > total_mem_per_cpu * 0.45:
+        if torch.cuda.device_count() > 1:
+            print("multi-gpu")
+            device_collection = [torch.device(i) for i in range(torch.cuda.device_count())]
+            model = auto_pipeline(model, device_collection, sample_inputs_tp)
+        else:
+            print("cpu")
+            model = model.cpu().float()
+    else:
+        print("single GPU")
+        model = model.cuda().half()
+
+    sample_inputs = []
+    for ints in sample_inputs_tp:
+        if type(ints) is torch.Tensor:
+            sample_inputs.append(ints.to(model.device))
+        else:
+            sample_inputs.append(ints)
+
+    input_keys, onnx_inputs = retrieve_onnx_inputs(model, sample_inputs)
+
+    onnx_path: Path = Path(onnx_path_str).absolute()
+    if onnx_path.suffix != ".onnx":
+        onnx_path = onnx_path / onnx_model_name
+
+    onnx_filepath_export_multi_files_tmp = onnx_path.parent / "tmp/tmp.onnx"
+    onnx_filepath_export_multi_files_tmp.parent.exists() and shutil.rmtree(onnx_filepath_export_multi_files_tmp.parent)
+    os.makedirs(onnx_filepath_export_multi_files_tmp.parent)
+
+    onnx_inp_names = ("input_ids", "attention_mask")
+    onnx_out_names = ("logits",)
+    onnx_dynamic_axes = {
+        "input_ids": {0: "batch_size", 1: "seq_len"},
+        "attention_mask": {0: "batch_size", 1: "seq_len"},
+    }
+    torch.onnx.export(
+        model=model,
+        args=onnx_inputs,
+        f=str(onnx_filepath_export_multi_files_tmp),
+        verbose=False,
+        opset_version=16,
+        input_names=onnx_inp_names,
+        output_names=onnx_out_names,
+        dynamic_axes=onnx_dynamic_axes,
+    )
+
+    onnx_model = onnx.load(str(onnx_filepath_export_multi_files_tmp))
+
+    onnx_path.exists() and onnx_path.unlink()
+    (onnx_path.parent / f"{onnx_model_name}_ext.data").exists() and (
+        onnx_path.parent / f"{onnx_model_name}_ext.data"
+    ).unlink()
```

Comment:
If this is needed, perhaps https://docs.python.org/3/library/pathlib.html#pathlib.Path.unlink:

```suggestion
    (onnx_path.parent / f"{onnx_model_name}_ext.data").unlink(missing_ok=True)
```